components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,331 @@
+  MDCActionSheetController *actionSheet = [[MDCActionSheetController alloc] initWithTitle:title
+                                                                                  message:message];
+  
+  return actionSheet;
+}
+
++ (instancetype)actionSheetControllerWithTitle:(NSString *)title {
+  return [MDCActionSheetController actionSheetControllerWithTitle:title message:nil];
+}
+
+- (instancetype)init {
+  return [MDCActionSheetController actionSheetControllerWithTitle:nil message:nil];
+}
+
+- (nonnull instancetype)initWithTitle:(NSString *)title message:(NSString *)message {
+  self = [super initWithNibName:nil bundle:nil];
+  if (self) {
+    _actionSheetTitle = [title copy];
+    _message = [message copy];
+    [self commonMDCActionSheetControllerInit];
+  }
+
+  return self;
+}
+
+- (void)commonMDCActionSheetControllerInit {
+  initialLayout = false;
+  _bottomSheet = [[MDCBottomSheetController alloc] initWithContentViewController:self];
+  _transitionController = [[MDCBottomSheetTransitionController alloc] init];
+  _transitionController.dismissOnBackgroundTap = YES;
+  super.transitioningDelegate = _transitionController;
+  super.modalPresentationStyle = UIModalPresentationCustom;
+  _actions = [[NSMutableArray alloc] init];
+  _tableView = [[MDCActionSheetListViewController alloc] initWithTitle:_actionSheetTitle
+                                                               message:_message
+                                                               actions:_actions];
+  _tableView.tableView.delegate = self;
+  [_tableView.tableView setTranslatesAutoresizingMaskIntoConstraints:NO];
+  _tableView.tableView.estimatedRowHeight = 56.f;
+  _tableView.tableView.rowHeight = UITableViewAutomaticDimension;
+  self.backgroundColor = [UIColor whiteColor];
+}
+
+- (void)addAction:(MDCActionSheetAction *)action {
+  [_tableView addAction:action];
+  [_tableView.tableView setNeedsLayout];
+  initialLayout = false;
+  [self.view setNeedsLayout];
+}
+
+- (NSArray<MDCActionSheetAction *> *)actions {
+  return _tableView.actions;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  
+  if (self.header == nil) {
+    self.header = [self headerView];
+  }
+
+  [self.view addSubview:_tableView.tableView];
+  [self.view addSubview:self.header];
+}
+
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+
+  if (initialLayout == false) {
+    [self firstLayout];
+  }
+}
+
+-(void)firstLayout {
+  [self.header setNeedsLayout];
+  [self.header layoutIfNeeded];
+
+  CGFloat width = CGRectGetWidth(self.view.bounds);
+  CGFloat height = CGRectGetHeight(self.header.frame) + [_tableView tableHeightForWidth:width];
+  CGRect tableFrame = _tableView.tableView.frame;
+  tableFrame.origin.y = CGRectGetHeight(self.header.frame);
+  _tableView.tableView.frame = tableFrame;
+  self.preferredContentSize = CGSizeMake(width, height);
+  initialLayout = true;
```

Comment:
Why is this `initialLayout` flag required? Please document its purpose.

Reply:
Added documentation within the source. This is needed because we are setting the `preferredContentSize` then after a resize of the application we can no longer get the frame to set the sheet height and width. No being able to set those properties correctly blocks us from layout out all other properties correctly.